sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v2/script.sql

```diff
@@ -0,0 +1,415 @@
+      WHERE
+        _table_suffix
+        BETWEEN FORMAT_DATE('%Y%m%d', DATE_SUB(@submission_date, INTERVAL 1 DAY))
+        AND FORMAT_DATE('%Y%m%d', @submission_date)
+        AND e.key = 'ga_session_id'
+        AND e.value.int_value IS NOT NULL
+      GROUP BY
+        a.user_pseudo_id,
+        CAST(e.value.int_value AS string)
+    ),
+    stub_session_ids_staging AS (
+      SELECT
+        user_pseudo_id AS ga_client_id,
+        event_timestamp,
+        CAST(
+          (
+            SELECT
+              `value`
+            FROM
+              UNNEST(event_params)
+            WHERE
+              key = 'ga_session_id'
+            LIMIT
+              1
+          ).int_value AS string
+        ) AS ga_session_id,
+        CAST(e.value.int_value AS string) AS stub_session_id
+      FROM
+        `moz-fx-data-marketing-prod.analytics_313696158.events_*`
+      JOIN
+        UNNEST(event_params) e
+      WHERE
+        _TABLE_SUFFIX
+        BETWEEN FORMAT_DATE('%Y%m%d', DATE_SUB(@submission_date, INTERVAL 1 DAY))
+        AND FORMAT_DATE('%Y%m%d', @submission_date)
+        AND event_name = 'stub_session_set'
+        AND e.key = 'id'
+        AND e.value.int_value IS NOT NULL
+    ),
+    last_stub_session_id AS (
+      SELECT
+        ga_client_id,
+        event_timestamp,
+        ga_session_id,
+        stub_session_id,
+        ROW_NUMBER() OVER (
+          PARTITION BY
+            ga_client_id,
+            ga_session_id
+          ORDER BY
+            event_timestamp DESC
+        ) AS stub_session_rnk
+      FROM
+        stub_session_ids_staging
+      QUALIFY
+        stub_session_rnk = 1
+    ),
+    all_stub_session_ids AS (
+      SELECT
+        ga_client_id,
+        ga_session_id,
+        ARRAY_AGG(stub_session_id) AS all_reported_stub_session_ids
+      FROM
+        stub_session_ids_staging
+      GROUP BY
+        ga_client_id,
+        ga_session_id
+    ),
+    landing_page_by_session_staging AS (
+      SELECT
+        user_pseudo_id AS ga_client_id,
+        CAST(
+          (
+            SELECT
+              `value`
+            FROM
+              UNNEST(event_params)
+            WHERE
+              key = 'ga_session_id'
+            LIMIT
+              1
+          ).int_value AS string
+        ) AS ga_session_id,
+        SPLIT(
+          (
+            SELECT
+              `value`
+            FROM
+              UNNEST(event_params)
+            WHERE
+              key = 'page_location'
+            LIMIT
+              1
+          ).string_value,
+          '?'
+        )[OFFSET(0)] AS page_location,
+        event_timestamp
+      FROM
+        `moz-fx-data-marketing-prod.analytics_313696158.events_*` a
+      JOIN
+        UNNEST(event_params) e
+      WHERE
+        _TABLE_SUFFIX
+        BETWEEN FORMAT_DATE('%Y%m%d', DATE_SUB(@submission_date, INTERVAL 1 DAY))
+        AND FORMAT_DATE('%Y%m%d', @submission_date)
+        AND e.key = 'entrances'
+        AND e.value.int_value = 1
```

Comment:
Why does this filter on this field?

Reply:
this is getting the first page view (i.e. entrance = 1) in order to get the landing page of the session